src/content/app/genome-browser/services/genome-browser-service/genomeBrowserCommands.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { EnsemblGenomeBrowser } from 'src/content/app/genome-browser/services/genome-browser-service/types/ensemblGenomeBrowser';
+
+export const setFocus = (payload: {
+  genomeBrowser: EnsemblGenomeBrowser;
+  focusId: string;
+  focusType: string;
+  genomeId: string;
+  bringIntoView?: boolean;
+}) => {
+  const { focusType } = payload;
+
+  if (focusType === 'gene') {
+    setFocusGene(payload);
+  } else if (focusType === 'variant') {
+    setFocusVariant(payload);
+  } else if (focusType === 'location') {
+    setFocusLocation(payload);
+  }
+};
+
```

Comment:
Isn't this a candidate for switch just in case more types get added?

Reply:
I dunno; there are all these `break`s in a switch, which you mustn't forget. It's just about as ugly as a bunch of `else if`s.